src/legacy/core_plugins/embeddable_api/public/get_actions_for_trigger.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Action } from './actions';
+import { IEmbeddable } from './embeddables';
+import { IContainer } from './containers';
+import { IRegistry, Trigger } from './types';
+
+export async function getActionsForTrigger(
```

Comment:
is this function part of API ? if so it should not be taking registries as parameters but rather import them internally

Reply:
This is done this way because this function can now be statically imported, but the registries hold state, so I think they should be accessed through the plugins module.   If it imported the registries directly, this function would need to be accessed on the plugin module.

Although all of this is really speculatively shaped based on NP semantics.  Are you okay punting this change to a post PR that shapes this plugin for NP migration?  It's not very consistent as it is right now, so it might be easier to focus on as the sole priority of a PR, e.g., obtaining the right NP shape/shims.